src/time.cc

```diff
@@ -57,17 +62,31 @@ std::string ToString(const std::chrono::system_clock::time_point& t) {
 
 std::chrono::system_clock::time_point FromString(const std::string& s) {
   std::tm tm;
-  const char* end = strptime(s.c_str(), "%Y-%m-%dT%H:%M:%S", &tm);
-  if (end == nullptr || end - s.c_str() != s.find('.')) {
-    // TODO
+  const char* end = strptime(s.c_str(), "%Y-%m-%dT%H:%M:", &tm);
+  if (end == nullptr || !std::isdigit(*end)) {
+    // TODO: Invalid time format.
+    return std::chrono::system_clock::time_point();
+  }
+  char* point;
+  long sec_i = std::strtol(end, &point, 10);
+  if ((point - end) != 2) {
+    // TODO: Seconds wasn't 2 digits.
     return std::chrono::system_clock::time_point();
   }
   char* zone;
-  long ns = std::strtol(end + 1, &zone, 10);
-  if (zone <= end + 1 || *zone != 'Z' || *(zone+1) != '\0') {
-    // TODO
+  double seconds = std::strtod(end, &zone);
+  if (sec_i < 0 || sec_i != static_cast<long>(seconds)) {
```

Comment:
Only minor note, I couldn't find a correlating unit test to this conditional statement. Every time there was a strange value in the place of the seconds, it was longer than 2 characters, that would have been caught by the previous condition. Might be overlooking the unit test that covers this case, just wanted to make a note.